azurerm/internal/services/network/tests/resource_arm_application_gateway_test.go

```diff
@@ -573,6 +573,25 @@ func TestAccAzureRMApplicationGateway_sslCertificate_keyvault(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMApplicationGateway_sslCertificate_EmptyPassword(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_application_gateway", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMApplicationGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApplicationGateway_sslCertificateEmptyPassword(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationGatewayExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
```

Comment:
this test is failing since the `ssl_certificate.0.data` (and `ssl_certificate.0.password`) fields are read from the config file, rather than the Azure API (as it doesn't return them) - meaning these don't exist at import time (since that doesn't read the config) as such we'll need to update this test to account for that:

```suggestion
			// since these are read from the existing state
			data.ImportStep(
				"ssl_certificate.0.data",
				"ssl_certificate.0.password",
			),
```

I hope you don't mind but since this otherwise LGTM I'm going to push a commit to get this fixed